activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -51,23 +51,21 @@ def select_one(arel, name = nil, binds = [])
 
       # Returns a single value from a record
       def select_value(arel, name = nil, binds = [])
-        arel, binds = binds_from_relation arel, binds
-        if result = select_rows(to_sql(arel, binds), name, binds).first
+        if result = select_rows(arel, name, binds).first
           result.first
         end
       end
 
       # Returns an array of the values of the first column in a select:
       #   select_values("SELECT id FROM companies LIMIT 3") => [1,2,3]
       def select_values(arel, name = nil, binds = [])
-        arel, binds = binds_from_relation arel, binds
-        select_rows(to_sql(arel, binds), name, binds).map(&:first)
+        select_rows(arel, name, binds).map(&:first)
       end
 
       # Returns an array of arrays containing the field values.
       # Order is the same as that returned by +columns+.
-      def select_rows(sql, name = nil, binds = [])
-        exec_query(sql, name, binds).rows
+      def select_rows(arel, name = nil, binds = [])
```

Comment:
I think the implementation so it doesn't have changes in the calls but but I prefer to keep all of them using `sql` than `arel` since `arel` is a implementation detail.